### PR TITLE
04 endpoint api articles

### PR DESCRIPTION
**Pull request by SO-1993**
This pull request implements the GET /api/articles/:article_id endpoint, which allows the retrieval of articles by their article_id. 

The PR ensures that the endpoint returns an article with the following properties: article_id, author, title, body, topic, created_at, votes, and article_img_url.

Additionally, error handling has been added to manage edge cases:
- 404 (invalid routes; article_id doesn't exist)
- 400 (nvalid article_id formats)

### CODE REVIEW

File: __tests__/articles-controller.test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+});
+
+describe("GET /api/articles/:article_id", () => {
+  test("should respond with a 200 status code when an article exists and has all required properties", () => {
+    const validArticleId = 1;
+    return request(app)
+      .get(`/api/articles/${validArticleId}`)
+      .expect(200)
+      .then((response) => {
+        expect(response.body.article).toHaveProperty(
```
Comment (LangaLee) [inline __tests__/articles-controller.test.js:20]: please test the actual article_id as you know what it should be

File: models/articles-models.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        votes,
+        article_img_url
+      FROM articles
+      WHERE article_id = $1`,
+      [article_id]
+    )
+    .then((result) => {
+      const article = result.rows[0];
+      if (!article) {
+        return null;
```
Comment (LangaLee) [inline models/articles-models.js:22]: Instead of returning null I would do return an error here and then this way you can pass the error to the next middleware and handle it there

**Review by LangaLee — COMMENTED** [review]
I left feedback below
